frontend/src/components/search/search.tsx

```diff
@@ -0,0 +1,90 @@
+import { SearchQueryParam } from 'common/enums/enums';
+import { FC, VideoCard as VideoCardType } from 'common/types/types';
+import { useAppDispatch, useAppSelector, useState, useCallback, useEffect, useSearchParams } from 'hooks/hooks';
+import { searchActions } from 'store/actions';
+import { FilterBar, FilterSidebar, VideoCard } from './components/components';
+
+import styles from './styles.module.scss';
+
+const Search: FC = () => {
+  const dispatch = useAppDispatch();
+  const {
+    search: { isFilterShown },
+  } = useAppSelector((state) => ({
+    search: state.search,
+  }));
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const [filterTypeValue, setFilterTypeValue] = useState('');
+  const [filterDateValue, setFilterDateValue] = useState('');
+  const [filterDurationValue, setFilterDurationValue] = useState('');
+  const [sortByValue, setSortByValue] = useState('');
+
+  const onHandleSetSearchParams = (prop: string, value: string): void => {
+    setSearchParams({ ...Object.fromEntries([...searchParams]), [prop]: value });
+  };
+
+  const onHandleChangeFilterType = (v: string): void => {
+    setFilterTypeValue(v);
+    onHandleSetSearchParams(SearchQueryParam.TYPE, v);
+  };
+
+  const onHandleChangeFilterDate = (v: string): void => {
+    setFilterDateValue(v);
+    onHandleSetSearchParams(SearchQueryParam.DATE, v);
```

Comment:
start searching should be triggered in useEffects once useEffect deps are updated.

Suggest to read values.filter(Boolean).join(separator);